src/uu/cp/src/cp.rs

```diff
@@ -1270,17 +1270,17 @@ fn copy_on_write_linux(source: &Path, dest: &Path, mode: ReflinkMode) -> CopyRes
     match mode {
         ReflinkMode::Always => unsafe {
             let result = ficlone(dst_file.as_raw_fd(), src_file.as_raw_fd() as *const i32);
-            if result != 0 {
-                return Err(format!(
+            return if result != 0 {
```

Comment:
This `return` can be removed and then the `Ok(())` moved to line 1289. That makes this the last expression of the function and makes it a little bit more intuitive.